### ISSUE TITLE
Cursor position is wrong with special characters

### ISSUE THREAD

**Issue by tgross35** [integer32llc/rust-playground#815]
Hello all,

I haven't been able to find this written up anywhere, so here's the issue. Special characters cause the position to offset, probably due to the Unicode helpers. See the example here:

https://play.rust-lang.org/?version=stable&mode=debug&edition=2021&gist=b7401426cca8ddcc8a53cc025adb6b4c

Place your cursor somewhere after the fancy `a` and type, you'll be typing one letter behind what you expect

**Comment by shepmaster** [comment]
Which editor?

![2A87E944-D4A6-4CE4-9292-051769D5DEA1](https://user-images.githubusercontent.com/174509/174594194-75feb28a-9ed3-4b94-bd41-163e965edb94.jpeg)

**Comment by tgross35** [comment]
@shepmaster you are correct, it's only on ace, noticed both on MacOS and Windows. Guess it's an easy fix to switch.

It is reproducable on that playground - just paste the `a̐` in. It displays correctly but seems like cursor position / editing is char-based rather than grapheme-based

**Comment by shepmaster** [comment]
You'll need to [file the issue with Ace, please](https://github.com/ajaxorg/ace).

**Comment by tgross35** [comment]
Thanks for the link, looks like here is the tracked issue https://github.com/ajaxorg/ace/issues/4142

Please close or leave open as you see fit
